### PR TITLE
Feat / GRW-1039 / CPAS tracking final

### PR DESCRIPTION
**Pull request by marian-buc**
## What?

- Removed quoteCartId from outside offer_data 
- Refactored sign_error event
- Added payment_details_confirmed event
- Refactored tracking hook to use async apolloClient.query instead of useQuery, removing race conditions and keeping the same API

## Why?
Overhauling tracking for CPAS 

**Ticket(s): [GRW-1039]**

[GRW-1039]: https://hedvig.atlassian.net/browse/GRW-1039?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/client/utils/tracking/trackOfferEvent.ts [hunk 1]
```diff
@@ -100,62 +105,51 @@ export const useTrackOfferEvent = () => {
...
-    (eventParams: EventParameters) => {
-      if (selectedBundleVariant) {
-        trackOfferCallback(eventParams, selectedBundleVariant)
-      } else {
-        setEventQueue((prevQueue) => [...prevQueue, eventParams])
+    ({ eventName, options = {} }: EventParameters) => {
+      const runQuery = async () => {
+        if (apolloClient && quoteCartId) {
+          try {
+            const quoteCartQuery = await apolloClient!.client.query<
```
Comment (robinandeer) [inline src/client/utils/tracking/trackOfferEvent.ts:113]: Why do we need the hashbang here if we do the if statement above?

File: src/client/utils/tracking/trackOfferEvent.ts [hunk 2]
```diff
@@ -100,62 +105,51 @@ export const useTrackOfferEvent = () => {
...
+              variables: {
+                id: quoteCartId,
+                locale: isoLocale,
+              },
+            })
+            return quoteCartQuery.data
+          } catch (e) {
+            return
+          }
+        } else return
```
Comment (robinandeer) [inline src/client/utils/tracking/trackOfferEvent.ts:126]: Guess you could handle this with "finally" block... but shouldn't we do something/log something is the query fails?

**Review by robinandeer — COMMENTED** [review]
Nice work on this. I had some comments - let me know if you wanna go over together.

I guess you discussed the updated event names with Sonny?

So with this approach, we skip the queue altogether but call the API/cache as soon as we call the API. I like it I think. The only thing is that this approach is not resilient if the query fails. Perhaps we should at least log that to datadog maybe?

**Comment by marian-buc** [comment]
@robinandeer @dendoesit addressed the err handling in https://github.com/HedvigInsurance/web-onboarding/pull/972

File: src/client/utils/tracking/trackOfferEvent.ts (hunk 1 above)
Comment (robinandeer) [reply src/client/utils/tracking/trackOfferEvent.ts:113]: @marian-buc bump ;)

Comment (marian-buc) [reply src/client/utils/tracking/trackOfferEvent.ts:113]: https://github.com/HedvigInsurance/web-onboarding/blob/0c888ab09114125b3ad47a5b7c76724a5e11d461/src/client/apolloClient.ts#L89

Query still might fail event if quoteCartId is supplied

Comment (robinandeer) [reply src/client/utils/tracking/trackOfferEvent.ts:113]: hmmm not sure what you mean there.. I was referring to the non-null assertion operator `!` here:

```
apolloClient!.client.query
```

Why is it still necessary? :)

**Review by robinandeer — APPROVED** [review]
Consider my comments on the non-null assertion operator -- otherwise I think it looks great!

**Review by robinandeer — APPROVED** [review]

File: src/client/utils/tracking/trackOfferEvent.ts (hunk 1 above)
Comment (marian-buc) [reply src/client/utils/tracking/trackOfferEvent.ts]: haha, yup, removed it in the new PR as well
[by passing the client as a param](https://github.com/HedvigInsurance/web-onboarding/pull/972/commits/9e89fe0c8438dd94f416448a38a84a8927907c26#diff-baa1c36a82a2c7876dc53d4df01709b66f9257464f8359b7f700d8e628abe6a5R32)

**Pull request merged**